File: 877a/main.py
import sys

input = lambda: sys.stdin.readline().rstrip()


def I():
    return input()


def II():
    return int(input())


def MII():
    return map(int, input().split())


def LI():
    return list(input().split())


def LII():
    return list(map(int, input().split()))

# alpha = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
s = I()
ans = 0
while "Danil" in s:
    ans += 1
    s = s.replace("Danil", '_', 1)
while "Olya" in s:
    ans += 1
    s = s.replace("Olya", '_', 1)
while "Slava" in s:
    ans += 1
    s = s.replace("Slava", '_', 1)
while "Ann" in s:
    ans += 1
    s = s.replace("Ann", '_', 1)
while "Nikita" in s:
    ans += 1
    s = s.replace("Nikita", '_', 1)

# print(s)
print("YES" if ans == 1 else "NO")

File: 877a/main.go
package main

import (
	"bufio"
	. "fmt"
	"io"
	"os"
	"strings"
)

func solve(in *bufio.Reader, out *bufio.Writer) {
	var b int
	Fscan(in, &b)
}

func run(r io.Reader, w io.Writer) {
	in := bufio.NewReader(r)
	out := bufio.NewWriter(w)
	defer out.Flush()
	var s string
	Fscan(in, &s)
	ans := 0
	for strings.Contains(s, "Danil") {
		ans += 1
		s = strings.Replace(s, "Danil", "_", 1)
	}
	for strings.Contains(s, "Olya") {
		ans += 1
		s = strings.Replace(s, "Olya", "_", 1)
	}
	for strings.Contains(s, "Slava") {
		ans += 1
		s = strings.Replace(s, "Slava", "_", 1)
	}
	for strings.Contains(s, "Ann") {
		ans += 1
		s = strings.Replace(s, "Ann", "_", 1)
	}
	for strings.Contains(s, "Nikita") {
		ans += 1
		s = strings.Replace(s, "Nikita", "_", 1)
	}
	if ans == 1 {
		Fprintln(out, "YES")
	} else {
		Fprintln(out, "NO")
	}
}

func main() {
	run(os.Stdin, os.Stdout)
}
